src/ECER.Clients.RegistryPortal/ECER.Clients.RegistryPortal.Server/PortalInvitations/PortalInvitationsEndpoints.cs

```diff
@@ -0,0 +1,33 @@
+﻿using ECER.Managers.Admin.Contract.PortalInvitations;
+using ECER.Managers.Registry.Contract.PortalInvitations;
+using ECER.Utilities.Hosting;
+using MediatR;
+using Microsoft.AspNetCore.Http.HttpResults;
+
+namespace ECER.Clients.RegistryPortal.Server.References;
+
+public class PortalInvitationsEndpoints : IRegisterEndpoints
+{
+  public void Register(IEndpointRouteBuilder endpointRouteBuilder)
+  {
+    endpointRouteBuilder.MapGet("/api/PortalInvitations/{token?}", async Task<Results<Ok<ReferenceQueryResult>, BadRequest<string>>> (string? token, IMediator messageBus, HttpContext httpContext, CancellationToken ct) =>
+    {
+      if (token == null) return TypedResults.BadRequest("Token is required");
+      var verificationResult = await messageBus.Send(new VerifyInviteTokenCommand(token), ct);
+      if (verificationResult == null || verificationResult.portalInvitation == Guid.Empty) return TypedResults.BadRequest("Invalid token");
+      // send portal invitation code and get reference details
+
+      var portalInvitationQueryResults = await messageBus.Send(new PortalInvitationQuery(verificationResult.portalInvitation), ct);
```

Comment:
No logic should exist in the endpoint - this is a use case and it needs to move into the manager